msprime/ancestry.py

```diff
@@ -1698,7 +1705,38 @@ class DiracCoalescent(ParametricAncestryModel):
 @dataclasses.dataclass
 class SweepGenicSelection(ParametricAncestryModel):
     """
-    .. todo:: Document me
+    This simulates a coalescent history with a single selective
+    sweep that has occurred in the past. The model is one of a
+    a structured coalescent where selective backgrounds are defined as in
+    `Braverman et al. (1995) <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1206652/>`_
+    The implementation details here follow closely to those in discoal,
+    `Kern and Schrider (2016)
+    <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5167068/>`_
+
+    .. warning::
+        As the strength of selection (:math:`2Ns`) increases the chance
+        of two coalescent events happening very very close together in
+        time and can lead to msprime errors. should you encounter this
+        decrease :math:`2Ns` if appropriate.
+
+    :param float position: the location of the beneficial allele along the
+        chromosome.
+    :param float start_frequency: population frequency of the benefical
+        allele at the start of the selective sweep. e.g., for a *de novo*
+        allele in a diploid population of size N, start frequency would be
+        :math:`1/2N`. A soft selective sweep could be modelled by
+        setting start_frequency :math:`> 1/2N`.
+    :param float end_frequency: population frequency of the beneficial
+        allele at the end of the selective sweep. For a complete sweep
+        this should be set to :math:`(2N-1)/2N`. For a partial sweep,
```

Comment:
Likewise, I don't think this is where we should go into the details of how to do particular types of sweeps. We should do a ``see :ref:`sec_ancestry_models_sweeps` for details on how to specify different types of sweeps. (Or maybe we have a subsection about partial/compete sweeps, etc. We don't have to *write* these sections now, just put in some placeholders and refer to them. That way at least the API docs are complete.)